src/Scrollyteller/index.tsx

```diff
@@ -30,6 +30,16 @@ const Scrollyteller = (props: Props) => {
 
   const base = useRef(null);
 
+  // Create and update onMarkerRef to make sure state inside
+  // the onMarker callback is up to date.
+  // This happens because useEffects attaching event listeners
+  // is executed only once, meaning that state inside callbacks (onMarker)
+  // will always have intial values.
+  const onMarkerRef = useRef(null);
+  useEffect(() => {
+    onMarkerRef.current = props.onMarker;
```

Comment:
Because `onMarkerRef.current` is updated on every re-render and called inside `onScroll`, didn't need to pass it to the deps array of useEffects